Matrix metrics

bcollica opened the pull request:
This PR adds functionality for computing additional summary metrics for estimators including the condition number, sign (PD, PSD, etc.), and the sparsity.  The main additions consist of two new functions in `summary_functions.R` along with minor updates and changes to other existing functions and tests.  The vignette has also been updated with a brief description of the new metrics.

Currently, the only supported plotting functionality for the new metrics is in the table of the best in class estimators displayed when `cvSummaryPlot` is called.  An additional PR is planned for updating some of the plotting functions, so I figured it would make the most sense to add functions for plotting the condition number (and possibly the sparsity) at that time.

bcollica committed:
Adding functions for calculating matrix metrics

bcollica committed:
Change cvMatrixMetrics to act as a "method" on cvCovEst object.

bcollica committed:
Update getHypers helper function

bcollica committed:
Reorder columns of matrix metrics dataframe

bcollica committed:
Add matrix metrics to summary functions and update docs.

bcollica committed:
Update summary call in plot functions

bcollica committed:
Update summary call in tests

bcollica committed:
Update summary call in vignette and mention the additional metrics.

bcollica committed:
Update documentation and vignette to reflect new matrix metrics and give descriptions.

PhilBoileau reviewed (approved):
This looks great! Once again, thank you for your fantastic work. There are only a few (very) minor changes to be made to some of the language in the docs (e.g. using "estimator" where "estimate" should be used instead). I'll make these changes and then merge.

PhilBoileau committed:
updating language in docs

PhilBoileau committed:
addressing note regarding undefined global vars

PhilBoileau committed:
updating pkgdown site

PhilBoileau committed:
bumping version num, ready for merge

PhilBoileau committed:
Merge branch 'master' into matrix_metrics

PhilBoileau committed:
forgot to update NEWS.md

PhilBoileau committed:
fixing merge conflicts

Merge branch 'matrix_metrics' of https://github.com/PhilBoileau/cvCovEst into matrix_metrics

# Conflicts:
#	docs/pkgdown.yml
